controllers/data.go

```diff
@@ -1,36 +1,73 @@
 package controllers
 
 import (
-	"html/template"
+	"encoding/json"
+	"fmt"
 	"net/http"
 
 	"deliverables/common/constants"
-	"deliverables/usecases"
+	"deliverables/entities"
 
 	"github.com/gorilla/mux"
 )
 
+type usecase interface {
+	GetAllPokemons() (map[int]entities.Pokemon, error)
+	GetPokemonById(id string) (map[int]entities.Pokemon, error)
+	GetPokemonFromAPI(id string) (map[int]entities.Pokemon, error)
+}
+
+type controller struct {
+	usecase usecase
+}
+
+func New(u usecase) controller {
+	return controller{u}
+}
+
 // Home - shows an hellow greet
-func Home(w http.ResponseWriter, r *http.Request) {
-	showData(w, constants.PageData{Message: "Hello World", Status: "Success"})
+func (c controller) Home(w http.ResponseWriter, r *http.Request) {
+	showData(w, constants.PageData{Message: "Hello World", Status: http.StatusOK})
 }
 
 //GetById - Function to get pokemon by ID
-func GetById(w http.ResponseWriter, r *http.Request) {
+func (c controller) GetById(w http.ResponseWriter, r *http.Request) {
 	vars := mux.Vars(r)
 	id := vars["id"]
 
-	data, _ := usecases.GetPokemonById(id)
-	showData(w, data)
+	data, err := c.usecase.GetPokemonById(id)
+
+	if err != nil {
+		showData(w, constants.PageData{Message: err.Error(), Status: http.StatusInternalServerError})
+	} else {
+		showData(w, constants.PageData{Data: data, Status: http.StatusOK})
+	}
 }
 
 //GetAll - Function to get all stored pokemons
-func GetAll(w http.ResponseWriter, r *http.Request) {
-	data, _ := usecases.GetAllPokemons()
-	showData(w, data)
+func (c controller) GetAll(w http.ResponseWriter, r *http.Request) {
+	data, err := c.usecase.GetAllPokemons()
+	if err != nil {
+		showData(w, constants.PageData{Message: err.Error(), Status: http.StatusInternalServerError})
+	} else {
+		showData(w, constants.PageData{Data: data, Status: http.StatusOK})
+	}
+}
+
+//GetAll - Function to get data from external API
+func (c controller) GetFromAPI(w http.ResponseWriter, r *http.Request) {
+	vars := mux.Vars(r)
+	id := vars["id"]
```

Comment:
You should validate the param
If is not valid or not set, return a `bad request`